pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -410,3 +415,194 @@ func enrichLegacyObject(originalObj, returnedObj runtime.Object) error {
 	accessorReturned.SetUID(accessorOriginal.GetUID())
 	return nil
 }
+
+func getSyncRequester(orgId int64) *identity.StaticRequester {
+	return &identity.StaticRequester{
+		Type:           identity.TypeServiceAccount, // system:apiserver
+		UserID:         1,
+		OrgID:          orgId,
+		Name:           "admin",
+		Login:          "admin",
+		OrgRole:        identity.RoleAdmin,
+		IsGrafanaAdmin: true,
+		Permissions: map[int64]map[string][]string{
+			orgId: {
+				"*": {"*"}, // all resources, all scopes
+			},
+		},
+	}
+}
+
+type syncItem struct {
+	name       string
+	objStorage runtime.Object
+	objLegacy  runtime.Object
+}
+
+func getList(ctx context.Context, obj rest.Lister, listOptions *metainternalversion.ListOptions) ([]runtime.Object, error) {
+	ll, err := obj.List(ctx, listOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	return meta.ExtractList(ll)
+}
+
+func mode2DataSyncer(ctx context.Context, legacy LegacyStorage, storage Storage, kind string, reg prometheus.Registerer, serverLockService ServerLockService, requestInfo *request.RequestInfo) (bool, error) {
+	metrics := &dualWriterMetrics{}
+	metrics.init(reg)
+
+	log := klog.NewKlogr().WithName("DualWriterMode2Syncer")
+
+	everythingSynced := false
+	outOfSync := 0
+	syncSuccess := 0
+	syncErr := 0
+
+	err := serverLockService.LockExecuteAndRelease(ctx, "dualwriter mode 2 sync", time.Minute*30, func(context.Context) {
+		log.Info("starting dualwriter mode 2 sync")
+		startSync := time.Now()
+
+		orgId := int64(1)
+
+		ctx = klog.NewContext(ctx, log)
+		ctx = identity.WithRequester(ctx, getSyncRequester(orgId))
+		ctx = request.WithNamespace(ctx, requestInfo.Namespace)
+		ctx = request.WithRequestInfo(ctx, requestInfo)
+
+		legacyList, err := getList(ctx, legacy, &metainternalversion.ListOptions{})
+		if err != nil {
+			log.Error(err, "unable to extract list from legacy storage")
+			return
+		}
+		log.Info("got items from legacy storage", "items", len(legacyList))
+
+		storageList, err := getList(ctx, storage, &metainternalversion.ListOptions{})
+		if err != nil {
+			log.Error(err, "unable to extract list from storage")
+			return
+		}
+		log.Info("got items from unified storage", "items", len(storageList))
+
+		itemsByName := map[string]syncItem{}
+		for _, obj := range legacyList {
+			accessor, err := utils.MetaAccessor(obj)
+			if err != nil {
+				log.Error(err, "error retrieving accessor data for object from legacy storage")
+				continue
+			}
+			name := accessor.GetName()
+
+			item, ok := itemsByName[name]
+			if !ok {
+				item = syncItem{}
+			}
+			item.name = name
+			item.objLegacy = obj
+			itemsByName[name] = item
+		}
+
+		for _, obj := range storageList {
+			accessor, err := utils.MetaAccessor(obj)
+			if err != nil {
+				log.Error(err, "error retrieving accessor data for object from storage")
```

Comment:
What do you think of adding the name to the log as well?
```suggestion
				log.WithValues("name", name).Error(err, "error retrieving accessor data for object from storage")
```

Reply:
oh! that's a good idea! Thanks for the suggestion!